src/Mod/Sketcher/Gui/Utils.cpp

```diff
@@ -315,39 +315,51 @@ void SketcherGui::ActivateHandler(Gui::Document* doc, DrawSketchHandler* handler
     }
 }
 
-bool SketcherGui::isCommandActive(Gui::Document* doc, bool actsOnSelection)
-{
+bool SketcherGui::isSketchInEdit(Gui::Document* doc) {
     if (doc) {
         // checks if a Sketch Viewprovider is in Edit and is in no special mode
-        if (doc->getInEdit()
-            && doc->getInEdit()->isDerivedFrom(SketcherGui::ViewProviderSketch::getClassTypeId())) {
-            auto mode =
-                static_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit())->getSketchMode();
-            if (mode == ViewProviderSketch::STATUS_NONE
-                || mode == ViewProviderSketch::STATUS_SKETCH_UseHandler) {
-                if (!actsOnSelection)
-                    return true;
-                else if (Gui::Selection().countObjectsOfType(
-                             Sketcher::SketchObject::getClassTypeId())
-                         > 0)
-                    return true;
+        auto* vp = dynamic_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit());
+        return (vp != nullptr);
+    }
+    return false;
+}
+
+bool SketcherGui::isCommandActive(Gui::Document* doc, bool actsOnSelection)
+{
+    if(isSketchInEdit(doc)) {
+        auto mode = static_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit())->getSketchMode();
+
+        if (mode == ViewProviderSketch::STATUS_NONE ||
+            mode == ViewProviderSketch::STATUS_SKETCH_UseHandler) {
+
+            if (!actsOnSelection) {
+                return true;
+            }
+            else if (Gui::Selection().countObjectsOfType(Sketcher::SketchObject::getClassTypeId()) > 0) {
+                return true;
             }
         }
     }
 
     return false;
 }
 
-SketcherGui::ViewProviderSketch* SketcherGui::getSketchViewprovider(Gui::Document* doc)
+SketcherGui::ViewProviderSketch* SketcherGui::getInactiveHandlerEditModeSketchViewProvider(Gui::Document* doc)
```

Comment:
I can't get the `InactiveHandler` part from th subsequent code. It seems to return a vp whatever is its handler state.

Reply:
The change of name comes to notify the user that it is only meaningful to call this function when the handler is inactive mode. When the handler is active, sketchgui data member is available so there is no need to call this function. Calling it incurs a performance penalisation in that case.